patterns/behavioural/RAII.md

```diff
@@ -108,6 +127,25 @@ Note that implementing `Deref` is not a core part of this pattern, it only makes
 using the guard object more ergonomic. Implementing a `get` method on the guard
 works just as well.
 
+When compared with RAII in C++, there are a few significant differences:
+
+* while C++ code often interfaces with C code or code in older styles, which doesn't use RAII, Rust was designed without the need to interface with such code, so its far less common to implement OBRM yourself
+* C++ doesn't have `Deref` nor a borrow checker, so code using RAII can not archive the same combination of safety and ergonomics
```

Comment:
> And then there's the whole issue of multi-threading...

could you elaborate/link some resource? Or are you simply referring to the general problem in C++, which presumably makes correct RAII harder too?

Reply:
Yes, I was simply referring to handling the problem in C++. And it's not just that you may need to keep re-entrant-safety in mind when working on some OBRM-implementation. Nothing stops you from placing a mutex _guard_/_lock_ into a `shared_ptr` and handing it out to multiple threads. Sure, it takes a certain kind of stupidity or ignorance to do so, but... well... humans do crazy stuff.
However, I wouldn't refer to that problem in this section because in Rust it's clearly irrelevant (safe for someone running around doing ridiculous `unsafe` stuff).

Reply:
So we fix that section by, depending on how well it fits, either also referencing operator* etc, or by removing that claim -  and perhaps also just moving that part to a seperate point - , the remainder is fine as is?

Reply:
I'd just drop the bit about `Deref`. The rest is fine imo.
```suggestion
* C++ doesn't have a borrow checker, so code using RAII can not archive the same combination of safety and ergonomics
```